build/upload_tarballs.jl

```diff
@@ -92,15 +64,9 @@ if abspath(PROGRAM_FILE) == @__FILE__
     # check that contents are tarballs with correct filename
     all(t -> !isnothing(match(TARBALL_REGEX, t)), readdir(TARBALL_DIR))
```

Comment:
first time I wrote this I wanted to see if there was a `file` command (similar to the bash command) that would assert the file type rather than check against a regex. I went looking in `Base.Filesystem` module but didn't see anything. Only calling it out in case you knew of one off hand - not suggesting we have to change it now.

Reply:
I think the usual way of handling this would be just to look at the extension. The other way is probably to use FileIO.jl and use the detection function which would detect the file type from magic bytes in the header: https://juliaio.github.io/FileIO.jl/stable/registry/#Registry-table